### PR TITLE
chore(admin): implement redux-toolkit

### PR DESCRIPTION
**Pull request by joshuaellis**
### What does it do?

* Lays the groundwork of introducing `redux-toolkit` in Strapi with types & typed hooks 👨🏼‍🍳 
* refactors to remove calls from `redux` and remove it from the deps (re-exported from `@reduxjs/toolkit`)
* refactors to remove calls from `reselect` and remove it from the deps (re-exported from `@reduxjs/toolkit`)

### Why is it needed?

* redux toolkit is the successor, it's easier to use with better features
* enables releases to use `@reduxjs/toolkit` from the start @markkaylor 

### Related issue(s)/PR(s)

* resolves CONTENT-1990

### CODE REVIEW

**Comment by gu-stav** [comment]
@joshuaellis waiting with a review until the FE tests are fixed ✌🏼

File: examples/getstarted/types/generated/contentTypes.d.ts
Comment (gu-stav) [inline examples/getstarted/types/generated/contentTypes.d.ts:1]: This change shouldn't be committed :)

File: packages/core/admin/admin/src/content-manager/pages/ListView/index.js [hunk 1]
```diff
@@ -880,27 +878,12 @@ ListView.propTypes = {
...
-      getDataSucceeded,
-      onChangeListHeaders,
-      onResetListHeaders,
-    },
-    dispatch
-  );
+  return bindActionCreators({}, dispatch);
 }
-const withConnect = connect(mapStateToProps, mapDispatchToProps);
+// const withConnect = connect(mapStateToProps, mapDispatchToProps);
```
Comment (gu-stav) [inline packages/core/admin/admin/src/content-manager/pages/ListView/index.js:887]: No longer needed?

**Review by gu-stav — COMMENTED** [review]
We probably should add redux-toolkit to the webpack alias file for now?

File: examples/getstarted/types/generated/contentTypes.d.ts
Comment (joshuaellis) [reply examples/getstarted/types/generated/contentTypes.d.ts:1]: I might add this to the .gitignore this is annoying me

**Comment by joshuaellis** [comment]
> We probably should add redux-toolkit to the webpack alias file for now?

I'd actually say no... I think we run the risk of breaking a plugin that perhaps uses an older version of redux-toolkit 🤔 imo lets move forward with the best practice and _not_?

**Pull request merged**